src/librustc_codegen_ssa/back/write.rs

```diff
@@ -803,8 +804,15 @@ fn execute_optimize_work_item<B: ExtraBackendMethods>(
 
     Ok(match lto_type {
         ComputedLtoType::No => {
-            let module = unsafe { B::codegen(cgcx, &diag_handler, module, module_config)? };
-            WorkItemResult::Compiled(module)
+            if !cgcx.opts.debugging_opts.combine_cgu
```

Comment:
Please deduplicate this entire `if` expression with the one below by putting it in a function

Reply:
Updated.